datatypes/src/raster/grid.rs

```diff
@@ -505,16 +505,31 @@ where
     }
 }
 
+impl<D, P> ByteSize for Grid<D, P>
+where
+    P: Pixel,
```

Comment:
why do we need P: Pixel? Wouldn't P: Sized also work?

Reply:
[15cc8a5](https://github.com/geo-engine/geoengine/pull/805/commits/15cc8a55c69d21b48e92135c7d0d40a3e7d4c5c9)